### PR TITLE
fix(compiler-cli): propagate preserveWhitespaces option to codegen

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**

### CODE REVIEW

File: packages/compiler-cli/src/codegen.ts [hunk 1]
```diff
@@ -103,6 +103,7 @@ export class CodeGenerator {
       locale: cliOptions.locale || undefined, missingTranslation,
       enableLegacyTemplate: options.enableLegacyTemplate !== false,
       enableSummariesForJit: options.enableSummariesForJit !== false,
+      preserveWhitespaces: options.preserveWhitespaces,
```
Comment (vicb) [inline packages/compiler-cli/src/codegen.ts]: any reason not to normalize here by adding the default value ? (as other options)

Comment (pkozlowski-opensource) [inline packages/compiler-cli/src/codegen.ts]: IMO it makes sense to normalize just in one place. In my mind CompilerOptions were objects that just carry what users set and CompilerConfig is the actual config based on options. Otherwise we start to have this normalization logic all over the place.

tl;dr; the proposed solution was, in my mind, centralizing logic in one place and this is why I'm not normalizing it here.

Comment (vicb) [inline packages/compiler-cli/src/codegen.ts]: Agree the normalization is probably a mess in the current code.

BUT is there any reason for this particular option to be different ?

I would prefer to see all options treated the same and a cleanup pass in an other PR for all the options. Do you think it is possible / makes sense here ?

Comment (pkozlowski-opensource) [inline packages/compiler-cli/src/codegen.ts]: If we take the "let's do this like for other options" argument as a deciding factor than I can change it. The only difference with the other options is that we _are_ planning to change it in the Angular 6.

tl;dr; is that the proposed change makes more sense to me but I also understand your arguments. Please make a call, I will update the PR accordingly.

Thnx for reviewing!

Comment (vicb) [inline packages/compiler-cli/src/codegen.ts]: If it makes more sense for you, let's go with it.

**Review by vicb — APPROVED** [review]
